src/NHibernate.Test/Hql/Ast/ParsingFixture.cs

```diff
@@ -1,176 +1,137 @@
-	//    {
-	//      XDocument doc = XDocument.Load(@"HQL Parsing\TestQueriesWithResults.xml");
-
-	//      foreach (XElement testGroup in doc.Element("Tests").Elements("TestGroup"))
-	//      {
-	//        string category = testGroup.Attribute("Name").Value;
-
-	//        foreach (XElement test in testGroup.Elements("Test"))
-	//        {
-	//          QueryTestData testData = new QueryTestData(category, test);
-
-	//          if (predicate(testData))
-	//          {
-	//            yield return projection(testData);
-	//          }
-	//        }
-	//      }
-	//    }
-
-	//    class QueryTestData
-	//    {
-	//      internal QueryTestData(string category, XElement xml)
-	//      {
-	//        Category = category;
-	//        Query = xml.Element("Query").Value;
-	//        Result = xml.Element("Result") != null ? xml.Element("Result").Value : "barf";
-	//        Name = xml.Element("Name") != null ? xml.Element("Name").Value : null;
-	//        Description = xml.Element("Description") != null ? xml.Element("Description").Value : null;
-	//        Ignore = xml.Attribute("Ignore") != null ? bool.Parse(xml.Attribute("Ignore").Value) : false;
-	//      }
-
-	//      internal string Category;
-	//      internal string Query;
-	//      internal string Result;
-	//      internal string Name;
-	//      internal string Description;
-	//      internal bool Ignore;
-	//    }
-	//  }
-	//}
+	[TestFixture]
+	public class ParsingFixture
+	{
+		/// <summary>
+		/// Key test for HQL strings -> HQL AST's - takes the query and returns a string
+		/// representation of the resultant tree
+		/// </summary>
+		/// <param name="query"></param>
+		/// <returns></returns>
+		[Test]
+		[TestCaseSource(typeof(QueryFactoryClass), nameof(QueryFactoryClass.TestCases))]
+		public string HqlParse(string query)
+		{
+			var p = new HqlParseEngine(query, false, null);
+			var result = p.Parse().ToStringTree();
+
+			return " " + result;
+		}
+
+		/// <summary>
+		/// Used to test individual queries "by hand", since td.net doesn't let me run a 
+		/// single test out of a data set
+		/// </summary>
+		[Test, Explicit]
+		public void ManualTest()
+		{
+			var p = new HqlParseEngine(@"select all s, s.Other from s in class Simple where s = :s", false, null);
+
+			var result = p.Parse().ToStringTree();
+
+			Console.WriteLine(result);
+		}
+
+		/// <summary>
+		/// Helper "test" to display queries that are ignored
+		/// </summary>
+		[Test, Explicit]
+		public void ShowIgnoredQueries()
+		{
+			foreach (string query in QueryFactoryClass.GetIgnores)
+			{
+				Console.WriteLine(query);
+			}
+		}
+
+		/// <summary>
+		/// Helper "test" to display queries that don't parse in H3
+		/// </summary>
+		[Test, Explicit]
+		public void ShowExceptionQueries()
+		{
+			foreach (string query in QueryFactoryClass.GetExceptions)
+			{
+				Console.WriteLine(query);
+			}
+		}
+
+		/// <summary>
+		/// Class used by Nunit 2.5 to drive the data into the HqlParse test
```

Comment:
Is that NUnit version information relevant? I doubt it.

```suggestion
		/// Class used by NUnit to drive the data into the HqlParse test.
```